docs/source/plugin-api/index.rst

```diff
@@ -0,0 +1,55 @@
+================
+Conda Plugin API
+================
+
+As of X.Y, Conda has support for user plugins, enabling them to extend and/or
+change some of its functionality.
+
+Plugins are implemented via pluggy_ and discovered using Python package
+entrypoints. You may refeer to pluggy_'s documentation for full description of
```

Comment:
Let's add a link to https://packaging.python.org/en/latest/specifications/entry-points/ for the Python entry points docs.